src/components/MCQ/MCQQuestion.vue

```diff
@@ -31,27 +31,43 @@ import { ref } from "vue";
 import type { MCQuestion, MCQOptions } from "@type/MCQ.d.ts";
 import MCQOption from "./MCQOption.vue";
 import MCQButton from "./MCQButton.vue";
+import { useQuizStore } from "@/store/QuizStore";
 
-const { statement, optionsList } = defineProps<MCQuestion>();
+const statUpdate = useQuizStore();
+
+const { statement, optionsList, _id } = defineProps<MCQuestion>();
 const selectedOption = ref<string | null>(null);
+const correct_index = ref<string | null>(null);
 const submitted = ref<boolean>(false);
 const emit = defineEmits(["nextQuestion", "skipQuestion"]);
 
 const submitAnswer = () => {
   submitted.value = true;
 };
 
-const nextQuestion = () => {
-  resetQuestion();
+const nextQuestion = (_id: { $oid: string }) => {
+  resetQuestion(_id);
   emit("nextQuestion");
 };
 
 const skipQuestion = () => {
-  resetQuestion();
+  resetQuestion(_id);
   emit("skipQuestion");
 };
 
-const resetQuestion = () => {
+//function to emit correct answer count
```

Comment:
We need to reduce adding comments, as Phil mentioned before, your function should be able to explain itself.

Reply:
fixed